app/models/spree/user.rb

```diff
@@ -47,6 +48,16 @@ def self.admin_created?
       User.admin.count > 0
     end
 
+    def self.from_omniauth(auth)
+      where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
+        user.provider = auth.provider
+        user.uid = auth.uid
+        user.email = auth.info.email
+        user.password = Devise.friendly_token[0,20]
+        user.confirmed_at ||= Time.now
```

Comment:
Here we automatically confirm the user if we're creating one, instead of waiting until they confirm via email; we may in fact want to wait until they click the link in their email.

Reply:
I think that makes sense. That makes me think of another test case for the specs; if we receive an invalid email address in the response from the server we'll want to handle some kind of validations and user feedback.